### ISSUE TITLE
Deleting a Product does not delete any of the associated Variant db rows

### ISSUE THREAD

**Issue by efatsi** [craftcms/commerce#147]
## Before:
```sql
mysql> SELECT * FROM craft_elements WHERE id = 1343;
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
| id   | fieldLayoutId | type                            | enabled | archived | dateCreated         | dateUpdated         | uid                                  |
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
| 1343 |           642 | craft\commerce\elements\Product |       1 |        0 | 2018-03-26 15:26:00 | 2018-03-26 18:13:16 | 6c0cd49a-f89a-4561-8252-4196c388a90f |
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
1 row in set (0.00 sec)

mysql> SELECT * FROM craft_elements WHERE id = 1344;
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
| id   | fieldLayoutId | type                            | enabled | archived | dateCreated         | dateUpdated         | uid                                  |
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
| 1344 |           643 | craft\commerce\elements\Variant |       1 |        0 | 2018-03-26 15:26:00 | 2018-03-26 18:13:16 | 26a4d360-853e-4d04-bea6-5b842da71a0e |
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
1 row in set (0.00 sec)

mysql> SELECT * FROM craft_commerce_variants WHERE sku = '86200';
+-----------+------+-------+-----------+---------+-----------+--------+--------+--------+--------+-------+-------------------+--------+--------+---------------------+---------------------+--------------------------------------+
| productId | id   | sku   | isDefault | price   | sortOrder | width  | height | length | weight | stock | hasUnlimitedStock | minQty | maxQty | dateCreated         | dateUpdated         | uid                                  |
+-----------+------+-------+-----------+---------+-----------+--------+--------+--------+--------+-------+-------------------+--------+--------+---------------------+---------------------+--------------------------------------+
|      1343 | 1344 | 86200 |         1 | 56.0000 |         1 | 0.0000 | 0.0000 | 0.0000 | 0.0000 |     0 |                 1 |   NULL |   NULL | 2018-03-26 15:26:00 | 2018-03-26 18:13:16 | 8852363f-34b8-4c2a-ab11-e681354a99ea |
+-----------+------+-------+-----------+---------+-----------+--------+--------+--------+--------+-------+-------------------+--------+--------+---------------------+---------------------+--------------------------------------+
1 row in set (0.00 sec)
```

## After deleting Product through CMS:
```sql
mysql> SELECT * FROM craft_elements WHERE id = 1343;
Empty set (0.00 sec)

mysql> SELECT * FROM craft_elements WHERE id = 1344;
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
| id   | fieldLayoutId | type                            | enabled | archived | dateCreated         | dateUpdated         | uid                                  |
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
| 1344 |           643 | craft\commerce\elements\Variant |       1 |        0 | 2018-03-26 15:26:00 | 2018-03-26 18:13:16 | 26a4d360-853e-4d04-bea6-5b842da71a0e |
+------+---------------+---------------------------------+---------+----------+---------------------+---------------------+--------------------------------------+
1 row in set (0.00 sec)

mysql> SELECT * FROM craft_commerce_variants WHERE sku = '86200';
+-----------+------+-------+-----------+---------+-----------+--------+--------+--------+--------+-------+-------------------+--------+--------+---------------------+---------------------+--------------------------------------+
| productId | id   | sku   | isDefault | price   | sortOrder | width  | height | length | weight | stock | hasUnlimitedStock | minQty | maxQty | dateCreated         | dateUpdated         | uid                                  |
+-----------+------+-------+-----------+---------+-----------+--------+--------+--------+--------+-------+-------------------+--------+--------+---------------------+---------------------+--------------------------------------+
|      NULL | 1344 | 86200 |         1 | 56.0000 |         1 | 0.0000 | 0.0000 | 0.0000 | 0.0000 |     0 |                 1 |   NULL |   NULL | 2018-03-26 15:26:00 | 2018-03-26 18:13:16 | 8852363f-34b8-4c2a-ab11-e681354a99ea |
+-----------+------+-------+-----------+---------+-----------+--------+--------+--------+--------+-------+-------------------+--------+--------+---------------------+---------------------+--------------------------------------+
1 row in set (0.00 sec)
```

**Comment by lukeholder** [comment]
Thanks for reporting.

Fixed by b113d516434

`composer update` to pull in the fixes.

**Comment by efatsi** [comment]
Thanks a lot!
